src/key.rs

```diff
@@ -916,27 +917,70 @@ impl XOnlyPublicKey {
     }
 }
 
-/// Opaque type used to hold the parity passed between FFI function calls.
+/// Represents the parity passed between FFI function calls.
 #[derive(Copy, Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
-pub struct Parity(i32);
+pub enum Parity {
+    /// Ever parity.
+    Even,
+    /// Odd parity.
+    Odd,
+}
```

Comment:
If you define
```
Even = 0,
Odd = 1,
```
then we can just cast this into a `i32` using `tweaked_parity as i32` idk if this casting is more or less preferred though (I know some people dislike this kind of C-like casts)

Reply:
I think we should support it, it's really convenient and we don't need to have an opinion about  whether it's good style.

Reply:
FTR the cast works without explicitly setting the variant values. I have however,  added the explicit variant values and removed the matching in `to_u8` in favour of casting. This puts the mapping of 0/1 -> Even/Odd in a single place only.

Reply:
Also, a benefit of the explicit values is not relying on the language implementation.